docs/source/model/python_model.rst

```diff
@@ -0,0 +1,437 @@
+
+Define a custom PythonModel
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Option 1: Subclass PythonModel
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:py:mod:`mlflow.pyfunc` module provides a :py:class:`generic PythonModel class <mlflow.pyfunc.PythonModel>` that can be used to define your own 
+customized model. By subclassing it, the model can be seamlessly integrated with other MLflow components.
+
+Methods of PythonModel:
+
+    - **predict**
+        A valid PythonModel must implement the predict method, which defines the model’s prediction logic. This method is called by MLflow when 
+        the model is loaded as a PyFunc model using ``mlflow.pyfunc.load_model`` and the ``predict`` function is invoked.
+    - **predict_stream**
+        The predict_stream method should be implemented if the model is intended for use in streaming environments. MLflow invokes this method 
+        when the model is loaded as a PyFunc model with ``mlflow.pyfunc.load_model`` and ``predict_stream`` is called.
+    - **load_context**
+        Implement the load_context method if the model requires additional context to be loaded. For more details, refer to :py:func:`load_context() <mlflow.pyfunc.PythonModel.load_context>`.
+
+.. tip::
+    Starting from MLflow 2.20.0, ``context`` parameter can be removed from ``predict`` and ``predict_stream`` functions if it is not used.
+    e.g. ``def predict(self, model_input, params)`` is a valid predict function signature.
+
+Below is an example of a simple PythonModel that takes a list of string and returns it.
+
+.. code-block:: python
+
+    import mlflow
+
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, model_input: list[str], params=None) -> list[str]:
+            return model_input
+
+
+
+Option 2: Define a callable
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+An alternative way to log a PythonModel is to define a callable that **takes a single argument** and returns a prediction. This callable can be
+logged as a PythonModel by passing it to ``mlflow.pyfunc.log_model``.
+
+.. tip::
+    Starting from MLflow 2.20.0, you can use @pyfunc decorator on the callable to enable data validation on the input based on the type hints.
+    Check :ref:`type hint usage in PythonModel <type-hint-usage-in-pythonmodel>` for more details.
+
+.. code-block:: python
+
+    def predict(model_input: list[str]) -> list[str]:
+        return model_input
+
+Log the model
+^^^^^^^^^^^^^
+PythonModel should be logged as python_function flavor using :py:func:`mlflow.pyfunc.log_model`.
+
+.. code-block:: python
+
+    import mlflow
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model",
+            python_model=MyModel(),
+            input_example=input_example,
+        )
+
+
+Validate the model before deployment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+After logging the model, you can validate the model by loading the model and running predictions. In addition, it's highly recommended to use 
```

Comment:
Let's lead with showing the example of using `mlflow.models.predict` in the top section and below mention that you can also locally (or remotely) load the logged model to validate in a new environment whether the model is ready for deployment. I'd show both examples. We should heavily promote the local subprocess with uv here in a distinct section.